[FLINK-18950][python][docs] Add documentation for Operations in Python DataStream API. (#13230)
apache/flink: docs/dev/python/datastream-api-users-guide/operators.md

DataStream programs in Flink are regular programs that implement transformations on data streams (e.g., mapping, 
filtering, reducing). Please see [operators]({% link dev/stream/operators/index.md %}
?code_tab=python) for an overview of the available stream transformations in Python DataStream API.

# Functions
Most operators require a user-defined function. The following will describe different ways of how they can be specified.

## Implementing Function Interfaces
Function interfaces for different operations are provided in Python DataStream API. Users can implement a Function 
interface and pass it to the corresponding operation. Take MapFunction for instance:
<p>
{% highlight python %}
# Implementing a MapFunction that returns plus one value of input value.
class MyMapFunction(MapFunction):
    
    def map(value):
        return value + 1
        
data_stream = env.from_collection([1, 2, 3, 4, 5],type_info=Types.INT())
mapped_stream = data_stream.map(MyMapFunction(), output_type=Types.INT())
{% endhighlight %}
</p>
<span class="label label-info">Note</span> In Python DataStream API, users are able to defined the output type information of the operation. If not 

> defined, the output type will be `Types.PICKLED_BYTE_ARRAY` so that data will be in a form of byte array generated by 

Reviewer: ```suggestion
specified, the output type will be `Types.PICKLED_BYTE_ARRAY` so that data will be in a form of byte array generated by 
```

pickle seriallizer. For more details about the `Pickle Serialization`, please refer to [DataTypes]({% link dev/python/datastream-api-users-guide/data_types.md
 %}#pickle-serialization).

## Lambda Functions

> As shown in the following example, all operations can also accept a lambda function to describe the operation:

Reviewer: ```suggestion
As shown in the following example, all the transformations can also accept a lambda function to define the functionality of the transformation:
```

<p>
{% highlight python %}

> data_stream = env.from_collection([1, 2, 3, 4, 5],type_info=Types.INT())

Reviewer: ```suggestion
data_stream = env.from_collection([1, 2, 3, 4, 5], type_info=Types.INT())
```

mapped_stream = data_stream.map(lambda x: x + 1, output_type=Types.INT())
{% endhighlight %}
</p>

> <span class="label label-info">Note</span> Operations ConnectedStream.map() and ConnectedStream.flat_map() are

Reviewer: ```suggestion
<span class="label label-info">Note</span> ConnectedStream.map() and ConnectedStream.flat_map() don't support lambda function and must accept `CoMapFunction` and `CoFlatMapFunction` separately. 
```

only support implementing CoMapFunction and CoFlatMapFunction Interface.

## Python Function

> Users can directly pass a python function to the operation as well:

Reviewer: ```suggestion
Users can also use Python function:
```

<p>
{% highlight python %}
def my_map_func(value):
    return value + 1

> data_stream = env.from_collection([1, 2, 3, 4, 5],type_info=Types.INT())

Reviewer: ```suggestion
data_stream = env.from_collection([1, 2, 3, 4, 5], type_info=Types.INT())
```

mapped_stream = data_stream.map(my_map_func, output_type=Types.INT())
{% endhighlight %}
</p>

> <span class="label label-info">Note</span> The arguements of the python function must be matched with the corresponding

Reviewer: this line seems unnecessary for me, isn't this obvious?
